lib/kitchen/driver/ec2.rb

```diff
@@ -98,10 +98,10 @@ class Ec2 < Kitchen::Driver::SSHBase
         end
       end
 
-      def self.iam_creds
+      def iam_creds
         @iam_creds ||= begin
-          fetch_credentials(use_iam_profile:true)
-        rescue RuntimeError => e
+          fetch_credentials(use_iam_profile: true)
+        rescue RuntimeError, NoMethodError => e
```

Comment:
NM, I was able to reproduce the error from https://github.com/test-kitchen/kitchen-ec2/pull/104/files locally and I see why you have the `NoMethodError` check